src/algorithms/calorimetry/CalorimeterHitRecoConfig.h

```diff
@@ -23,8 +23,7 @@ namespace eicrecon {
     double                   thresholdValue{0};
 
     // sampling fraction
-    double                   sampFrac{1.0};
-    std::vector<double>      sampFracLayer{};
+    std::string              sampFrac{"1.0"};
```

Comment:
Can we handle this as follows?
```suggestion
    std::variant<double,std::string> sampFrac{1.0};
```

Main benefit would actually be for expansion of this pattern. If we start to roll this out as everywhere, but it it requires us to change defaults to runtime-evaluated strings, we lose the compile-time validity checks on numerical values.

Reply:
Just to make sure you don't think you need to keep this open on my behalf, I'm ok moving away from the variant approach if it doesn't work technically, but I am worried about what we lose when we stick everything in strings including constants.